Presentation/Program.cs

```diff
@@ -0,0 +1,500 @@
+
+                Console.Clear();
+
+                if (choiceString == "back") break;
+
+                removeFriendReq = CheckAcceptOrRemove(choiceString);
+
+                if (removeFriendReq) choiceString = choiceString[1..];
+
+                choice = ConvertInputToInt(choiceString);
+
+                try
+                {
+                    Task<Unit> command = mediator.Send(new ValidateIdFriend
+                    {
+                        LoggedUser = loggedUser,
+                        idFriend = choice
+                    });
+
+                    if (command.IsFaulted) throw command.Exception;
+
+                    mediator.Send(new AcceptOrRemoveFriendRequest
+                    {
+                        LoggedUser = loggedUser,
+                        idFriend = choice,
+                        removeFriendRequest = removeFriendReq
+                    });
+
+                    if (removeFriendReq == true)
+                    {
+                        Console.WriteLine("Request removed succesfully.");
+                    }
+                    else
+                    {
+                        Console.WriteLine("Request accepted succesfully.");
+                    }
+                    Console.WriteLine();
+                    break;
+                }
+                catch (AggregateException ex)
+                {
+                    Console.WriteLine(ex.InnerException.Message + "\n");
+                }
+            }
+        }
+
+        public static void AddFriendMenu(User loggedUser, IMediator mediator)
+        {
+            int choice;
+            string? choiceString;
+
+            while (true)
+            {
+                Console.Write("Introduce the ID of the person you want to add: ");
+
+                choiceString = Console.ReadLine();
+
+                Console.Clear();
+
+                if (choiceString == "back") break;
+
+                choice = ConvertInputToInt(choiceString);
+
+                try
+                {
+                    Task<Unit> command = mediator.Send(new ValidateIdFriend
+                    {
+                        LoggedUser = loggedUser,
+                        idFriend = choice
+                    });
+
+                    if (command.IsFaulted) throw command.Exception;
+
+                    mediator.Send(new SendFriendRequest
+                    {
+                        LoggedUser = loggedUser,
+                        idFriend = choice
+                    });
+
+                    Console.WriteLine("Request sent successfully.\n");
+                    break;
+
+                }
+                catch (AggregateException ex)
+                {
+                    Console.WriteLine(ex.InnerException.Message + "\n");
+                }
+            }
+        }
+
+        public static void ValidateDeleteFriendId(User loggedUser, int choice, IMediator mediator)
+        {
+            if (loggedUser.Id == choice)
```

Comment:
Don't you check for this in the repository already?

Reply:
I have an operation called ValidateIdFriend that's similar to this function but throws a UserInFriendsException when the id of the selected user is in the logged user's friends list. 
I found a workaround for that. Check "adapted code to use ValidateIdFriend in DeleteFriendMenu" commit